internal/resources/appplatform/dashboard_resource.go

```diff
@@ -0,0 +1,182 @@
+package appplatform
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/grafana/grafana/apps/dashboard/pkg/apis/dashboard/v1alpha1"
+	"github.com/hashicorp/terraform-plugin-framework-jsontypes/jsontypes"
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+)
+
+// DashboardSpecModel is a model for the dashboard spec.
+type DashboardSpecModel struct {
+	JSON  jsontypes.Normalized `tfsdk:"json"`
+	Title types.String         `tfsdk:"title"`
+	Tags  types.List           `tfsdk:"tags"`
+}
+
+// Dashboard creates a new Grafana Dashboard resource.
+func Dashboard() resource.Resource {
+	return NewResource(ResourceConfig[*v1alpha1.Dashboard, *v1alpha1.DashboardList, v1alpha1.DashboardSpec]{
+		Kind: v1alpha1.DashboardKind(),
+		Schema: ResourceSpecSchema{
+			Description: "Manages Grafana dashboards.",
+			MarkdownDescription: `
+Manages Grafana dashboards via the new Grafana App Platform API. This resource is currently **EXPERIMENTAL** and may be subject to change. It requires a development build of Grafana with specific feature flags enabled.
+	`,
+			DeprecationMessage: "This resource is currently EXPERIMENTAL and may be subject to change.",
+			SpecAttributes: map[string]schema.Attribute{
+				"json": schema.StringAttribute{
+					Required:    true,
+					Description: "The JSON representation of the dashboard spec.",
+					CustomType:  jsontypes.NormalizedType{},
+				},
+				"title": schema.StringAttribute{
+					Optional:    true,
+					Description: "The title of the dashboard. If not set, the title will be derived from the JSON spec.",
+				},
+				"tags": schema.ListAttribute{
+					Optional:    true,
+					Description: "The tags of the dashboard. If not set, the tags will be derived from the JSON spec.",
+					ElementType: types.StringType,
+				},
+			},
+		},
+		SpecParser: func(ctx context.Context, spec types.Object, dst *v1alpha1.Dashboard) diag.Diagnostics {
+			var data DashboardSpecModel
+			if diag := spec.As(ctx, &data, basetypes.ObjectAsOptions{
+				UnhandledNullAsEmpty:    true,
+				UnhandledUnknownAsEmpty: true,
+			}); diag.HasError() {
+				return diag
+			}
+
+			var res v1alpha1.DashboardSpec
+			if diag := data.JSON.Unmarshal(&res); diag.HasError() {
+				return diag
+			}
+
+			if !data.Title.IsNull() && !data.Title.IsUnknown() {
+				res.Object["title"] = data.Title.ValueString()
+			}
+
+			if tags, ok := getTagsFromModel(data); ok {
+				res.Object["tags"] = tags
+			}
+
+			delete(res.Object, "version")
```

Comment:
Assuming that is done for the same reason as in `SpecSaver`

```suggestion
      // HACK: for v0 we need to clean a few fields from the spec,
			// which are not supposed to be set by the user.
			delete(res.Object, "version")
```

Reply:
Added, thank you!